client/wfprev-war/src/main/angular/src/app/components/edit-project/endorsement-approval/endorsement-approval.component.ts

```diff
@@ -0,0 +1,178 @@
+    MatDatepickerModule,
+    MatButtonModule,
+    DetailsContainerComponent,
+    CheckboxComponent,
+    DatePickerComponent,
+    ReadOnlyFieldComponent,
+    TextareaComponent
+  ],
+  templateUrl: './endorsement-approval.component.html',
+  styleUrl: './endorsement-approval.component.scss'
+})
+export class EndorsementApprovalComponent implements OnChanges {
+
+  @Input() fiscal!: ProjectFiscal;
+  @Input() currentUser!: string;
+  @Output() saveEndorsement = new EventEmitter<ProjectFiscal>(); 
+
+  endorsementApprovalForm = new FormGroup({
+    endorseFiscalYear: new FormControl<boolean | null>(false),
+    endorsementDate: new FormControl<Date | null>(null),
+    endorsementComment: new FormControl<string | null>(''),
+    approveFiscalYear: new FormControl<boolean | null>(false),
+    approvalDate: new FormControl<Date | null>(null),
+    approvalComment: new FormControl<string | null>(''),
+  });
+
+  ngOnInit(): void {
+    this.endorsementApprovalForm.get('endorseFiscalYear')?.valueChanges.subscribe((checked) => {
+      if (checked) {
+        this.endorsementDateControl.setValue(new Date());
+      } else {
+        this.endorsementDateControl.setValue(null);
+      }
+    });
+
+    this.endorsementApprovalForm.get('approveFiscalYear')?.valueChanges.subscribe((checked) => {
+      if (checked) {
+        this.approvalDateControl.setValue(new Date());
+      } else {
+        this.approvalDateControl.setValue(null);
+      }
+    });
+  }
+
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes['fiscal']?.currentValue) {
+      const fiscal = changes['fiscal'].currentValue as ProjectFiscal;
+      const endorseChecked = !!fiscal.endorserName;
+      const approveChecked = !!fiscal.isApprovedInd;
+
+      this.endorsementApprovalForm.patchValue({
+        endorseFiscalYear: endorseChecked,
+        endorsementDate: endorseChecked && fiscal.endorsementTimestamp
+          ? new Date(fiscal.endorsementTimestamp)
+          : null,
+        endorsementComment: fiscal.endorsementComment ?? '',
+        approveFiscalYear: approveChecked,
+        approvalDate: approveChecked && fiscal.approvedTimestamp
+          ? new Date(fiscal.approvedTimestamp)
+          : null,
+        approvalComment: fiscal.businessAreaComment ?? ''
+      });
+
+      this.endorsementApprovalForm.markAsPristine();
+    }
+  }
+
+
+
+  get effectiveEndorserName(): string {
+    const checked = this.endorsementApprovalForm.get('endorseFiscalYear')?.value;
+    return checked ? this.currentUser : (this.fiscal?.endorserName || '');
+  }
+
+
+  onSave() {
+    const formValue = this.endorsementApprovalForm.value;
+
+    const endorsementRemoved = !formValue.endorseFiscalYear;
+    const approvalRemoved = !formValue.approveFiscalYear;
+    const currentStatusCode = this.fiscal?.planFiscalStatusCode?.planFiscalStatusCode;
+    const shouldResetToPrepared =
+      (endorsementRemoved || approvalRemoved) &&
+      currentStatusCode !== 'DRAFT' &&
```

Comment:
It's generally a good idea to put these kind of code values like `DRAFT`, `PROPOSED` etc into constants in case they need to be used in multiple areas or change in the future. it's much easier to find them in a centralized location vs hunting and pecking around in the code.

Reply:
I see Sukh made the same comment, so ignore this one!